src/app/job-editor/utilities/path-resolver.ts

```diff
@@ -1,25 +1,41 @@
 const path = require("path");
 
 export function ensureAbsolutePaths(rootPath: string, job: Object): Object {
-    const parsed = {...job};
 
-    Object.keys(parsed).forEach(key => {
-        const val = parsed[key];
+    return Object.keys(job).reduce((acc, key) => {
+        return Object.assign(acc, {[key]: deepEnsureAbsolutePaths(job[key], rootPath)});
+    }, {});
+}
+
+function deepEnsureAbsolutePaths(entry, rootPath) {
+    if (Array.isArray(entry)) {
+        return entry.map(value => deepEnsureAbsolutePaths(value, rootPath));
+    } else if (Object.prototype.isPrototypeOf(entry)) {
 
-        const isFileOrDir = val && val.class && ~["File", "Directory"].indexOf(val.class);
+        if (!isFileOrDir(entry)) {
+            return Object.keys(entry).reduce((acc, key) =>
+                Object.assign(acc, {[key]: deepEnsureAbsolutePaths(entry[key], rootPath)}), {}
+            );
+        }
 
-        if (isFileOrDir) {
-            if (val.path && !isAbsolutePath(val.path)) {
-                val.path = path.normalize(rootPath + path.sep + val.path);
-            }
+        const updated = {...entry};
+        if (updated.path && !isAbsolutePath(updated.path)) {
```

Comment:
path.resolve(rootPath, updated.path) instead of path.normalize(rootPath + path.sep + updated.path).

If its ok then no need for isAbsolutePath(updated.path) condition.